File: frontend/src/types/case.ts
export interface Review {
    id: number;
    case_id: number;
    status: 'Accepted' | 'Edited' | 'Rejected';
    reason?: string;
    created_at: string;
}

export interface Case {
    id: number;
    case_id: string;
    category: string;
    difficulty: string;
    diagnosis_type: string;
    symptom: string;
    topology_note: string;
    show_outputs: string;
    expected_fault: string;
    osi_layer: string;
    concept_tag: string;
    severity: string;
    expected_next_command?: string;
    expected_fix?: string;
    verification_command?: string;
    
    // AI Diagnosis results
    ai_root_cause?: string;
    ai_osi_layer?: string;
    ai_confidence?: string;
    ai_evidence?: string;
    ai_reason?: string;
    ai_next_command?: string;
    ai_fix_steps?: string;
    ai_verification_command?: string;
    
    // Stateful fields
    diagnosis_status?: 'NEEDS_INFO' | 'DIAGNOSED' | 'PENDING_REVIEW' | 'VERIFICATION_REQUIRED' | 'RESOLVED' | 'NOT_RESOLVED' | 'REJECTED';
    
    // Optional review from relationships
    review?: Review;
}

export interface RuleFinding {
    rule: string;
    status: 'DETECTED' | 'NOT_DETECTED' | 'INSUFFICIENT_EVIDENCE' | 'NOT_APPLICABLE';
    evidence?: string[] | string;
    [key: string]: any;
}

export interface DiagnosisResponse {
    status: 'NEEDS_INFO' | 'DIAGNOSED' | 'VERIFICATION_REQUIRED' | 'RESOLVED' | 'NOT_RESOLVED';
    root_cause?: string;
    osi_layer?: string;
    confidence?: string;
    evidence?: string;
    reason?: string;
    next_command?: string;
    fix_steps?: string;
    verification_command?: string;
    rule_findings?: RuleFinding[];
}

export interface DashboardStats {
    issues_by_type: { name: string; value: number }[];
    issues_by_severity: { name: string; value: number }[];
    review_stats: {
        total_reviews: number;
        accepted: number;
        edited: number;
        rejected: number;
        agreement_rate: number;
    };
}

File: backend/app/schemas/case.py

from pydantic import BaseModel

from app.schemas.review import ReviewResponse


class CaseBase(BaseModel):
    case_id: str
    category: str
    difficulty: str
    diagnosis_type: str
    symptom: str
    topology_note: str
    show_outputs: str
    expected_fault: str
    osi_layer: str
    concept_tag: str
    severity: str
    expected_next_command: str | None = None
    expected_fix: str | None = None
    verification_command: str | None = None

class CaseCreate(BaseModel):
    symptom: str
    topology_note: str
    show_outputs: str
    
    # Optional fields for dynamic cases
    category: str | None = "Dynamic"
    difficulty: str | None = "Unknown"
    diagnosis_type: str | None = "Dynamic"
    
    # Ground truth fields are not expected for dynamic ingestion
    expected_fault: str | None = "Unknown"
    osi_layer: str | None = "Unknown"
    concept_tag: str | None = "Dynamic"
    severity: str | None = "Unknown"
    expected_next_command: str | None = None
    expected_fix: str | None = None
    verification_command: str | None = None

class CaseResponse(CaseBase):
    id: int
    
    ai_root_cause: str | None = None
    ai_osi_layer: str | None = None
    ai_confidence: str | None = None
    ai_evidence: str | None = None
    ai_reason: str | None = None
    ai_next_command: str | None = None
    ai_fix_steps: str | None = None
    ai_verification_command: str | None = None
    
    review: ReviewResponse | None = None

    model_config = {"from_attributes": True}
